lib/isEmail.js

```diff
@@ -175,10 +175,23 @@ function isEmail(str, options) {
       }
     }
   }
+  /**
+  * Check the first or last string it ends with special characters.
+  * Also, check if special characters apear consecutively.
+  */
+
+
+  var regEndsWithSpecialChar = /^.*[!#$%&',*+-\/=?^_`{|]$/i;
+  var regStartsWithSpecialChar = /^[!#$%&',*+-\/=?^_`{|]/i;
+  var regConsecutiveSpecialChar = /([!#$%&',*+-\/=?^_`{|])\1/i;
```

Comment:
@profnandaa is there a way to make this regex validate `abc_+kl@gmail.com`. For now, it validates `abc__dc@mail.com`. 
In short, what I mean is that I have caught most of the case of invalid emails, except an instance where when two instances of special characters that are different e.g `-+` occur concurrently. 
See example above.